GitUI/CommandsDialogs/FormDeleteRemoteBranch.cs

```diff
@@ -52,58 +61,74 @@ private void FormDeleteRemoteBranchLoad(object sender, EventArgs e)
             }
         }
 
-        private void OkClick(object sender, EventArgs e)
+        protected override void OnShown(EventArgs e)
+        {
+            RecalculateSizeConstraints();
+            base.OnShown(e);
+            Branches.Focus();
+        }
+
+        private void RecalculateSizeConstraints()
         {
-            try
+            SuspendLayout();
+            MinimumSize = MaximumSize = Size.Empty;
+
+            int height = ControlsPanel.Height + MainPanel.Padding.Top + MainPanel.Padding.Bottom
+                       + tlpnlMain.Height + tlpnlMain.Margin.Top + tlpnlMain.Margin.Bottom + DpiUtil.Scale(42);
+
+            MinimumSize = new Size(tlpnlMain.PreferredSize.Width + DpiUtil.Scale(70), height);
+            MaximumSize = new Size(Screen.PrimaryScreen.Bounds.Width, height);
+            Size = new Size(Width, height);
+            ResumeLayout();
+        }
+
+        private void Delete_Click(object sender, EventArgs e)
+        {
+            if (!DeleteRemote.Checked)
+            {
+                return;
+            }
+
+            List<IGitRef> selectedBranches = Branches.GetSelectedBranches().ToList();
+
+            bool hasUnmergedBranches = selectedBranches.Any(branch => !_mergedBranches.Contains(branch.CompleteName));
+            if (hasUnmergedBranches)
```

Comment:
```suggestion
            if (hasUnmergedBranches && !AppSettings.DontConfirmDeleteUnmergedBranch)
```

Reply:
No, this operation is a potential data loss scenario, a confirmation must not be suppressed.